server/sandbox_run.go

```diff
@@ -217,6 +225,9 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 
 	// add labels
 	labels := req.GetConfig().GetLabels()
+
+	// Add special container name label for the infra container
+	labels[KubernetesContainerNameLabel] = PodInfraContainerName
```

Comment:
Mmm I'm pretty sure we're overriding the old value provided by kube and this might break tests